File: round_3b_live_coding/deepflow-interview-chatbot/contracts/python/chat_contracts/messages.py
import uuid
from datetime import UTC, datetime
from enum import StrEnum

from pydantic import BaseModel, Field


class Role(StrEnum):
    USER = "user"
    ASSISTANT = "assistant"


class ChatMessage(BaseModel):
    """A single chat message persisted in the database."""

    id: str = Field(default_factory=lambda: str(uuid.uuid4()))
    conversation_id: str
    role: Role
    content: str
    thinking_content: str | None = Field(
        default=None,
        description="Raw reasoning/thinking content produced by the model, if any.",
    )
    created_at: datetime = Field(default_factory=lambda: datetime.now(UTC))

File: round_3b_live_coding/deepflow-interview-chatbot/contracts/typescript/src/messages.ts
import { z } from "zod";

export const Role = z.enum(["user", "assistant"]);
export type Role = z.infer<typeof Role>;

export const ChatMessage = z.object({
  id: z.string().uuid(),
  conversation_id: z.string(),
  role: Role,
  content: z.string(),
  thinking_content: z.string().nullable().optional(),
  created_at: z.string().datetime(),
});
export type ChatMessage = z.infer<typeof ChatMessage>;
